Sources/ClientRuntime/Networking/Streaming/BufferedStream.swift

```diff
@@ -0,0 +1,124 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// A `Stream` implementation that buffers data in memory.
+/// The buffer size depends on the amount of data written and read.
+/// Note: This class is thread-safe.
+public class BufferedStream: Stream {
+
+    /// The length of the stream, if known.
```

Comment:
The underlying buffer is not nullable and so what case(s) is the length unknown?

Reply:
answered in some other thread.